src/FakeItEasy/Expressions/ExpressionArgumentConstraintFactory.cs

```diff
@@ -77,7 +80,7 @@ private static object InvokeExpression(Expression expression)
             return Expression.Lambda(expression).Compile().DynamicInvoke();
         }
 
-        private IArgumentConstraint GetArgumentConstraintFromExpression(Expression expression)
+        private IArgumentConstraint GetArgumentConstraintFromExpression(Expression expression, out object value)

```

Comment:
In fact, I was trying to refactor the code to pass a `ParsedArgumentExpression` to `GetArgumentConstraintFromExpression`, which would make the code simpler, but unfortunately we don't have a `ParsedArgumentExpression` in `CreateParamArrayConstraint`...